tortoisebeacon/tortoise_beacon.go

```diff
@@ -642,19 +637,23 @@ func (tb *TortoiseBeacon) runConsensusPhase(ctx context.Context, epoch types.Epo
 }
 
 func (tb *TortoiseBeacon) markProposalPhaseFinished(epoch types.EpochID) {
+	finishedAt := time.Now()
+
 	tb.proposalPhaseFinishedTimestampsMu.Lock()
```

Comment:
can have an RLock here

Reply:
@antonlerner Why? We write here, not read

Reply:
my bad misread it